src/lib/marlin_plonk_bindings/stubs/src/pasta_vesta.rs

```diff
@@ -1,92 +1,103 @@
+use crate::arkworks::{CamlFp, CamlFq, CamlGVesta, CamlGroupProjectiveVesta};
+use ark_ec::{AffineCurve, ProjectiveCurve};
+use ark_ff::{One, UniformRand};
 use mina_curves::pasta::{
-    vesta::{Affine as GAffine, Projective as GProjective},
     fp::Fp,
     fq::Fq,
-};
-use algebra::{
-    curves::{AffineCurve, ProjectiveCurve},
-    One, UniformRand,
+    vesta::{Affine as GAffine, Projective},
 };
 use rand::rngs::StdRng;
 
 #[ocaml::func]
-pub fn caml_pasta_vesta_one() -> GProjective {
-    GProjective::prime_subgroup_generator()
+pub fn caml_pasta_vesta_one() -> CamlGroupProjectiveVesta {
+    Projective::prime_subgroup_generator().into()
 }
 
 #[ocaml::func]
 pub fn caml_pasta_vesta_add(
-    x: ocaml::Pointer<GProjective>,
-    y: ocaml::Pointer<GProjective>,
-) -> GProjective {
-    (*x.as_ref()) + (*y.as_ref())
+    x: ocaml::Pointer<CamlGroupProjectiveVesta>,
+    y: ocaml::Pointer<CamlGroupProjectiveVesta>,
+) -> CamlGroupProjectiveVesta {
+    x.as_ref() + y.as_ref()
 }
 
 #[ocaml::func]
 pub fn caml_pasta_vesta_sub(
-    x: ocaml::Pointer<GProjective>,
-    y: ocaml::Pointer<GProjective>,
-) -> GProjective {
-    (*x.as_ref()) - (*y.as_ref())
+    x: ocaml::Pointer<CamlGroupProjectiveVesta>,
+    y: ocaml::Pointer<CamlGroupProjectiveVesta>,
+) -> CamlGroupProjectiveVesta {
+    x.as_ref() - y.as_ref()
 }
 
 #[ocaml::func]
-pub fn caml_pasta_vesta_negate(x: ocaml::Pointer<GProjective>) -> GProjective {
-    -(*x.as_ref())
+pub fn caml_pasta_vesta_negate(
+    x: ocaml::Pointer<CamlGroupProjectiveVesta>,
+) -> CamlGroupProjectiveVesta {
+    -x.as_ref()
 }
 
 #[ocaml::func]
-pub fn caml_pasta_vesta_double(x: ocaml::Pointer<GProjective>) -> GProjective {
-    x.as_ref().double()
+pub fn caml_pasta_vesta_double(
+    x: ocaml::Pointer<CamlGroupProjectiveVesta>,
+) -> CamlGroupProjectiveVesta {
+    x.as_ref().double().into()
 }
 
 #[ocaml::func]
-pub fn caml_pasta_vesta_scale(x: ocaml::Pointer<GProjective>, y: Fp) -> GProjective {
-    x.as_ref().mul(y)
+pub fn caml_pasta_vesta_scale(
+    x: ocaml::Pointer<CamlGroupProjectiveVesta>,
+    y: ocaml::Pointer<CamlFp>,
```

Comment:
Why has this type changed?

Reply:
hum... I can't remember, but I remember having trouble making this work with the new `mul()` in arkworks